source/adapters/opencl/command_buffer.cpp

```diff
@@ -148,11 +158,18 @@ UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferAppendKernelLaunchExp(
 
   cl_command_properties_khr *Properties =
       hCommandBuffer->IsUpdatable ? UpdateProperties : nullptr;
+
+  cl_sync_point_khr *RetSyncPoint =
+      hCommandBuffer->IsInOrder ? nullptr : pSyncPoint;
+  const cl_sync_point_khr *SyncPointWaitList =
+      hCommandBuffer->IsInOrder ? nullptr : pSyncPointWaitList;
+  uint32_t WaitListSize =
+      hCommandBuffer->IsInOrder ? 0 : numSyncPointsInWaitList;
```

Comment:
Nit but would prefer to store `hCommandBuffer->IsInOrder` in a local variable rather than calling that 3 times. Also applies to other functions in this file that use the same pattern.